homeassistant/components/matter/binary_sensor.py

```diff
@@ -161,4 +161,100 @@ def _update_from_device(self) -> None:
         entity_class=MatterBinarySensor,
         required_attributes=(clusters.DoorLock.Attributes.DoorState,),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmDeviceMutedSensor",
+            measurement_to_ha=lambda x: x
+            == clusters.SmokeCoAlarm.Enums.MuteStateEnum.kMuted,
+            translation_key="muted",
+            icon="mdi:bell-off",
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.DeviceMuted,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmEndfOfServiceSensor",
+            measurement_to_ha=lambda x: x
+            == clusters.SmokeCoAlarm.Enums.EndOfServiceEnum.kExpired,
+            translation_key="end_of_service",
+            device_class=BinarySensorDeviceClass.PROBLEM,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.EndOfServiceAlert,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmBatteryAlertSensor",
+            measurement_to_ha=lambda x: x
+            != clusters.SmokeCoAlarm.Enums.AlarmStateEnum.kNormal,
+            translation_key="battery_alert",
```

Comment:
Does this mean that the battery is low? In that case you can omit the translation key and it will fallback to using the device class name and translation

Reply:
With other words: We need the it to be named "Battery Alert" or "Batter Low Alert".
The device also has a battery sensor attached on the powersource cluster (in case of a battery powered device):

![Scherm­afbeelding 2024-09-24 om 14 47 40](https://github.com/user-attachments/assets/09a067fb-e237-48b4-8653-e3cb90f171c1)

Reply:
The power source cluster is not mandatory for smoke sensor, that is why we put battery alert attribute in smoke cluster as well. 

User must replace a new batter once the battery alert is set.

Both of Heiman smoke sensor and CO sensor are supported power source cluster and the batter alert attribute in Smoke cluster.